### PR TITLE
[No ticket] Joe Sandbox Add optional filename parameter to submit_sample

### PR DESCRIPTION
**Pull request by pmara-r7**
## Proposed Changes

Add optional filename parameter to submit_sample action, add additional dictionary param that is defaulting to the incorrect value after an API update

### Description

Describe the proposed changes:

  - Add file type parameter to the submit sample action. This is needed in cases where Joe Sandbox can't automatically detect the filetype.
  - Add no "hybrid-decompilation" when submitting sample. I saw it in testing this action. Might effect other actions, if it does, check out the solution in this PR- worked for me. The API indicate it should be defaulting to False, but that seems to not be the case right now

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

**Comment by pmara-r7** [comment]
## Assessment
### Run

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nConnect: Connected successfully\npj-test/Joe https://example.com Step name: submit_sample\n",
    "meta": {},
    "output": {
      "webids": [
        "2253765"
      ]
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/joe_sandbox:1.0.4 run < tests/submit_sample.json
</summary>
</details>

File: plugins/joe_sandbox/help.md [hunk 1]
```diff
@@ -32,9 +32,7 @@ Example input:
 
 ```
 {
-  "api_key": {
-      "secretKey": "8e8786182c66e8bc2abdab9198f1385691987bfe2a4917be1268e915e457dbc5"
-  },
+  "api_key": "8e8786182c66e8bc2abdab9198f1385691987bfe2a4917be1268e915e457dbc5",
```
Comment (mrinehart-r7) [inline plugins/joe_sandbox/help.md:35]:
```suggestion
  "api_key": "30f800f97aeaa8d62bdf3a6fb2b0681179a360c12e127f07038f8521461e5050",
```

File: plugins/joe_sandbox/plugin.spec.yaml [hunk 2]
```diff
@@ -239,6 +239,12 @@ actions:
         type: bytes
         required: true
         example: "TVqQAAMAAAAEAAAA//8AALgAAAAAAA..."
+      filename:
+        title: Filename
+        description: Used to give Joe Sandbox a hint at what file type we are uploading. Extension (ex .txt) required
```
Comment (mrinehart-r7) [inline plugins/joe_sandbox/plugin.spec.yaml:244]:
```suggestion
        description: Used to give Joe Sandbox a hint at what file type is being uploaded. File extension (eg .txt, .zip) required
```

File: plugins/joe_sandbox/komand_joe_sandbox/actions/submit_sample/action.py [hunk 3]
```diff
@@ -35,6 +39,11 @@ def run(self, params={}):
             raise Exception(
                 'Unable to decode base64 input for "cookbook". ' "Contents of the file must be encoded with base64!"
             )
+        webids = None
+        if filename:
+            sample_tuple = (filename, sample_bytes)
```
Comment (mrinehart-r7) [inline plugins/joe_sandbox/komand_joe_sandbox/actions/submit_sample/action.py:44]: No need to change it now, but for future, there's no need to include the `tuple` part of the variable name - we know it's a tuple by looking at it :) Just `sample` would suffice.

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]
Small comments & regenerate should fix it up!

**Review by mberezin-r7 — APPROVED** [review]

**Review by mrinehart-r7 — APPROVED** [review]

**Pull request merged**